extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -116,98 +116,81 @@ class AmpViqeoPlayer extends AMP.BaseElement {
         )
         ||
         (this.kindIsProd_
           ? 'https://cdn.viqeo.tv/js/vq_player_init.js'
           : 'https://static.viqeo.tv/js/vq_player_init.js?branch=dev1'
         );
+    // embed preview url
+    let previewUrl = this.element.getAttribute('data-player-url');
+    previewUrl =
+        (previewUrl
+            && previewUrl.length && decodeURI(previewUrl)
+        )
+        || (this.kindIsProd_ ? 'https://cdn.viqeo.tv/embed' : 'https://stage.embed.viqeo.tv');
 
-    const ampIframe = this.element.ownerDocument.createElement('iframe');
-    ampIframe.src = 'about:blank';
-    if (ampIframe.contentWindow) {
-      frameLoaded.call(this);
-    } else {
-      ampIframe.onload = frameLoaded.bind(this);
-    }
-
-    this.element.appendChild(ampIframe);
-    this.iframe_ = ampIframe;
-
-    return this.loadPromise(ampIframe);
-
-    /** @this {AmpViqeoPlayer} */
-    function frameLoaded() {
-      const doc = ampIframe.contentWindow.document;
-      setStyles(doc.body, {
-        'marginLeft': 0,
-        'marginRight': 0,
-        'marginTop': 0,
-        'marginBottom': 0,
-      });
-
-      const scr = doc.createElement('script');
-      scr.async = true;
-      scr.src = scriptPlayerInit;
-      doc.head.appendChild(scr);
-
-      const mark = doc.createElement('div');
-
-      setStyles(mark, {
-        'position': 'relative',
-        'width': '100%',
-        'height': '0',
-        'paddingBottom': '100%',
-      });
-      mark.setAttribute('data-vnd', this.videoId_);
-      mark.setAttribute('data-profile', this.profileId_);
-      mark.classList.add('viqeo-embed');
-
-      const iframe = doc.createElement('iframe');
-
-      iframe.setAttribute('width', iframeWidth);
-      iframe.setAttribute('height', iframeHeight);
-      iframe.setAttribute('style', iframeStyle);
-      iframe.setAttribute('frameBorder', '0');
-      iframe.setAttribute('allowFullScreen', '');
-      iframe.src = this.getVideoIframeSrc_();
-
-      mark.appendChild(iframe);
-
-      const wrapper = doc.createElement('div');
-      wrapper.appendChild(mark);
-
-      doc.body.appendChild(wrapper);
-
-      const ampIframeWindow = ampIframe.contentWindow;
+    // Create preview iframe source path
+    previewUrl = assertAbsoluteHttpOrHttpsUrl(
+        `${previewUrl}/?vid=${this.videoId_}`);
+
+    const jsonParams = {
+      scriptPlayerInit: encodeURIComponent(scriptPlayerInit),
+      previewUrl: encodeURIComponent(previewUrl),
+      videoId: this.videoId_,
+      profileId: this.profileId_,
+    };
+
+    const iframe = getIframe(this.win, this.element, 'viqeoplayer', jsonParams);
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleViqeoMessages_.bind(this)
+    );
+
+    return this.mutateElement(() => {
+      this.element.appendChild(iframe);
+      this.iframe_ = iframe;
+      this.applyFillContent(iframe);
+      this.getViewport().onScroll(this.sendIntersectionRatio_.bind(this));
```

Comment:
@sdbaron We already support `autoplay` in AMP that behaves the same way and include some builtin UX. All you need to do is register with `video-manager`. We leave the control to page authors, if they put `autoplay` on the player, then it kicks in.

Reply:
Thanks! That's what I need!